server.go

```diff
@@ -573,6 +573,13 @@ func (srv *Server) serveDNS(m []byte, w *response) {
 		req.Ns, req.Answer, req.Extra = nil, nil, nil
 		req.Zero = false
 
+		w.WriteMsg(req)
+		fallthrough
```

Comment:
This double fallthrough doesn't work. This would be my suggestion:

```go
switch action := srv.MsgAcceptFunc(dh); action {
case MsgAccept:
	// ... exsting code ...
	fallthrough
case MsgReject, MsgNotImplemented:
	if accept == MsgNotImplemented {
		req.SetRcodeNotImplemented(req)
	} else {
		req.SetRcodeFormatError(req)
	}
	req.Ns, req.Answer, req.Extra = nil, nil, nil
	// ... existing code ...
	fallthrough
case MsgIgnore:
	// ... existing code ...
	return
}
```